cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -185,6 +185,95 @@ The diagram for this code would be:
    :width: 8in
    :align: center
 
+Conditional Node
+----------------
+
+You can provide a *conditional* node in your structure that allows for branching based on 
+a boolean predicate.
+
+Taking our first example and modifying it, you could use code such as::
+
+  public class ConditionalWorkflow extends AbstractWorkflow {
+
+    @Override
+    public void configure() {
+      setName("ConditionalWorkflow");
+      setDescription("Demonstration of conditional execution of a Workflow");
+      
+      addMapReduce("JoinWithCatalogMR");
+      
+      condition(new MyPredicate())
+        .addMapReduce("BuildProductProfileMR")
+      .otherwise()
+        .addMapReduce("BuildUserProfileMR")
+      .end();
+      
+      addAction(new UploadProfilesCA());
+    }
+  }
+
+where ``MyPredicate`` is a public class which implements the ``Predicate`` interface as::
+
+  public static class MyPredicate implements Predicate<WorkflowContext> {
+
+    @Override
+    public boolean apply(@Nullable WorkflowContext input) {
+      if (input != null && input.containsKey("BuildProductProfile")) {
+        Map<String, Long> customCounters = input.getToken().getMapReduceCounters().get("MyCustomCounters");
+        if (customCounters.get("BuildProductProfile") > 0) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+  
+In the ``JoinWithCatalogMR`` MapReduce, it could have in its Mapper class code that 
```

Comment:
In the ``JoinWithCatalogMR`` MapReduce, custom Hadoop counters can be emitted that